### PR TITLE
Review: v1.9.3 to master

### PR DESCRIPTION
**Pull request by Jack-Works**
## Big Changes
- Refactor on AdditionalContent 1886ee5e9e808e76d7e3f568ef3357d990534a0c
- Remove WelcomBanner c8c953b7b2dff59c6e5f97d705aefad87747b83d

## Notes
- Add type coverage 161d5c3a8e0afa1e582f20cf985e6eb259e25238
- Add red packet to storybook & refactor it 859613d2c793ce253a1daa9a74ecd537d4bbc40f 335aca656ac82228ed6be76021bf7850db427332 5c3cbdc5732fdf5bd59d51fe77dde03e7e3b6f1d b899d41f4e18afdf19481b925332c485bb5894b4 689a76c73c6781178a18ec4cbd2c623edffb6459 7b8c13c39fa73c6cf014d8a278ddba8a206a2778 3445b829fcecc7c1ce0aa2c59ca906727000161c
- Refactor on a3c9b402a4160360a1c68d0f6a44efb668b3f5cb

## Discussions
1. We can recognize text in the link now, but it increase the code complicity. Switch to a markdown library might be better ( @neruthes )
3. <del>Why `shouldDisplayWelcome` deprecated? ( @SunriseFox  )</del>

## Todos
- [x] src/components/InjectedComponents/PostDialog.tsx , `hasRedPacket` should become a MaterialUI theme. ( @SunriseFox ? )
- [x] Maybe rename src/components/InjectedComponents/StructuredMessage/StructuredPluginWrapper.tsx
- [x] Add red packet related UI to storybook ( @SunriseFox ? )
- [x] Split `useSettingsUI` out to a separate file
- [x] Check the enum support for `useSettingsUI`
- [x] Change the design of MessageCenter in holoflows-kit (@Jack-Works )

## Future todos:
- [ ] Make SteganographyService more configurable (?)
- [ ] Remove the spring festival special leading text (https://github.com/DimensionDev/Maskbook/commit/81b5c8a4c4745b7189f85fe2243754fc1362041d#diff-e07e5a7386945a884d7b9bbc226204a6R339 ) ( but when? @neruthes )
- [ ] Resolve this todo https://github.com/DimensionDev/Maskbook/blob/v1.10.0/src/database/Persona/helpers.ts#L155 in `createPersonaByMnemonic`? ( @SunriseFox , @guanbinrui  )

### CODE REVIEW

**Comment by guanbinrui** [comment]
> How to resolve this todo https://github.com/DimensionDev/Maskbook/blob/v1.10.0/src/database/Persona/helpers.ts#L155 in createPersonaByMnemonic? ( @SunriseFox , @guanbinrui )

Use message center?

**Comment by SunriseFox** [comment]
`shouldDisplayWelcome`: as our setup procedure changes, a user will always gain access permission first and then starts immersive setup  (dashboard->provider instead of provider->dashboard, no longer need to input username from provider and open dashboard). 

We need a new design about what to prompt if a user deletes their persona / disconnects all their profiles. @neruthes

File: src/components/InjectedComponents/PostDialog.tsx [hunk 1]
```diff
@@ -147,11 +147,15 @@ export function PostDialogUI(props: PostDialogUIProps) {
...
                             <Chip
-                                onDelete={() => {
+                                onDelete={async () => {
                                     const ref = getActivatedUI().typedMessageMetadata
                                     const next = new Map(ref.value.entries())
                                     next.delete('com.maskbook.red_packet:1')
                                     ref.value = next
+                                    if (props.onShareToEveryoneChanged) {
+                                        await sleep(300)
+                                        props.onShareToEveryoneChanged(false)
```
Comment (Jack-Works) [inline src/components/InjectedComponents/PostDialog.tsx]: why add a delay here?

Comment (guanbinrui) [reply src/components/InjectedComponents/PostDialog.tsx]: For resetting `Share to Everyone` clip to unselect state. In addition, `ref.value = next` is async so wait a while for ensure really reseted.

Comment (Jack-Works) [reply src/components/InjectedComponents/PostDialog.tsx]: maybe add a comment here

**Comment by Jack-Works** [comment]
> `shouldDisplayWelcome`: as our setup procedure changes, a user will always gain access permission first and then starts immersive setup (dashboard->provider instead of provider->dashboard, no longer need to input username from provider and open dashboard).
> 
> We need a new design about what to prompt if a user deletes their persona / disconnects all their profiles. @neruthes

Should we remove all codes that related to shouldDisplayWelcome

**Comment by neruthes** [comment]
> @Jack-Works: We can recognize text in the link now, but it increase the code complicity. Switch to a markdown library might be better

What text in what link? Could you offer additional context?

----

> @Jack-Works: Remove the spring festival special leading text (81b5c8a#diff-e07e5a7386945a884d7b9bbc226204a6R339 ) ( but when? @neruthes )

Let us consult @Tedko for an answer.

Also, I would like to learn your ideas about <https://github.com/DimensionDev/Maskbook-Talks/issues/22>.

----

> @SunriseFox: We need a new design about what to prompt if a user deletes their persona / disconnects all their profiles.

The behavior has been defined in #407.

Upon loading Dashboard: If there is no Persona currently, Dashboard will be in pre-initialization state; otherwise, Dashboard will be in post-initialization state. This selection is made every time when Dashboard is loaded.

Nothings explicitly happens when the the user deletes all Personas. Next time the user loads Dashboard, the Dashboard will be displayed as Pre-Init state.

----

**Comment by SunriseFox** [comment]
> > @SunriseFox: We need a new design about what to prompt if a user deletes their persona / disconnects all their profiles.
> 
> The behavior has been defined in #407.
> 
> Upon loading Dashboard: If there is no Persona currently, Dashboard will be in pre-initialization state; otherwise, Dashboard will be in post-initialization state. This selection is made every time when Dashboard is loaded.
> 
> Nothings explicitly happens when the the user deletes all Personas. Next time the user loads Dashboard, the Dashboard will be displayed as Pre-Init state.

No. I am talking about what if 
1. we have permissions to manipulate facebook / twitter page 
2. the user has no persona and never opens dashboard

**Comment by Tedko** [comment]
> 
> > @Jack-Works: Remove the spring festival special leading text ([81b5c8a](https://github.com/DimensionDev/Maskbook/commit/81b5c8a4c4745b7189f85fe2243754fc1362041d)#diff-e07e5a7386945a884d7b9bbc226204a6R339 ) ( but when? @neruthes )
> 
> Let us consult @Tedko for an answer.
> 
> Also, I would like to learn your ideas about [DimensionDev/Maskbook-Talks#22](https://github.com/DimensionDev/Maskbook-Talks/issues/22).
> 


We can revise these text from 'Happy New Year etc' to 'Best Wish' etc, and remove hashtags . I guess we can do an update about leading text around 元宵节 -- ping me again when you are ready to publish a new version

**Comment by neruthes** [comment]
> > > @SunriseFox: We need a new design about what to prompt if a user deletes their persona / disconnects all their profiles.
> > 
> > 
> > The behavior has been defined in #407.
> > Upon loading Dashboard: If there is no Persona currently, Dashboard will be in pre-initialization state; otherwise, Dashboard will be in post-initialization state. This selection is made every time when Dashboard is loaded.
> > Nothings explicitly happens when the the user deletes all Personas. Next time the user loads Dashboard, the Dashboard will be displayed as Pre-Init state.
> 
> No. I am talking about what if
> 
> 1. we have permissions to manipulate facebook / twitter page
> 2. the user has no persona and never opens dashboard

Suggestion for initialization can be implemented in a way similar to #409, to be visible when the user visits a timeline page of a supported social network. I will note this for future reference.

**Pull request merged**